bin/journey-test.sh

```diff
@@ -233,15 +233,15 @@ solve_all_exercises() {
 
     pushd ${exercism_exercises_dir}/${TRACK}/${exercise}
     # Check that tests compile before we strip @Ignore annotations
-    "$EXECPATH"/gradlew compileTestJava
+    "$EXECPATH"/gradlew compileTestJava --configure-on-demand --parallel
```

Comment:
[This](http://logans.me/profile/profile-2017-07-07-21-49-23.html) is the profile generated of when I run `../gradlew test --profile` while in the exercises directory. Configuring the projects took ~13 seconds and the entire length it ran was 3:31.
[This](http://logans.me/profile/profile-2017-07-07-21-54-09.html) is the profile generated from the same command, but I added `--configure-on-demand`. Configuration took about 6 seconds instead here and the total build time was about 2:55 :)